lib/measured/measurable.rb

```diff
@@ -6,7 +6,7 @@ class Measured::Measurable < Numeric
   def initialize(value, unit)
     raise Measured::UnitError, "Unit value cannot be blank" if value.blank?
 
-    @unit = self.class.unit_system.to_unit_name!(unit)
+    @unit = unit.is_a?(Measured::Unit) ? unit : self.class.unit_system.unit_for!(unit)
```

Comment:
I really like the consistency of having one way to just say `Turn this input into the object I want` and use it throughout the file. You can see if it's a big perf impact.